src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputFormattedUseOfOptionalBraces.java

```diff
@@ -0,0 +1,293 @@
+package com.google.checkstyle.test.chapter4formatting.rule411optionalbracesusage;
+
+class InputFormattedUseOfOptionalBraces {
+  /**
+   * some javadoc..
+   *
+   * @return helper func *
+   */
+  boolean condition() {
+    return false;
+  }
+
+  /** Test do/while loops. */
+  void testDoWhile() {
+    // Valid
+    do {
+      testDoWhile();
+    } while (condition());
+
+    // Invalid
+    do testDoWhile(); // violation ''do' construct must use '{}'s.'
+    while (condition());
+  }
+
+  /** Test while loops. */
+  void testWhile() {
+    // Valid
+    while (condition()) {
+      testWhile();
+    }
+
+    // Invalid
+    while (condition())
+      ; // violation above ''while' construct must use '{}'s.'
+    while (condition()) // violation ''while' construct must use '{}'s.'
+    testWhile(); // violation ''while' child has incorrect indentation level 4, expected .* 6.'
```

Comment:
So, checkstyle is right here and in other similar cases below

Reply:
This can be resolved by formatter, I am not sure why they did not target it. 
Let's recheck with formatter by creating issue on them

Reply:
We should treat violations on formatting after formatter as red frag on our misinterpretation of style.

Reply:
there are 2 bugs of formatter here:

1. No usage of `{}` ( https://github.com/google/google-java-format/issues/51 )
2. Incorrect indentation for 'while' child ( https://github.com/google/google-java-format/issues/1009, this is a similar issue to our problem )

The issues already seems to be reported to them

Reply:
lets keep this discussion unresovled to ease finding reasons why there are violation later on.